autorest/codegen/models/base_schema.py

```diff
@@ -23,6 +23,7 @@ def __init__(self, namespace: str, yaml_data: Dict[str, Any]) -> None:
         self.default_value = yaml_data.get("defaultValue", None)
         self.xml_metadata = yaml_data.get("serialization", {}).get("xml", {})
         self.api_versions = set(value_dict["version"] for value_dict in yaml_data.get("apiVersions", []))
+        self.nullable = yaml_data.get("extensions", {}).get("x-nullable")
```

Comment:
Why here? All schema can be nullable?

Reply:
Based on [this](https://github.com/Azure/autorest/tree/master/docs/extensions#x-nullable), it looks like all schemas can be nullable except arrays. Still going to keep this code here so I don't have to duplicate it across all of the different schemas, but will override in ListSchema so self.nullable there is always False